src/stream/src/executor/flow_control.rs

```diff
@@ -58,10 +58,12 @@ impl FlowControlExecutor {
                         let result = rate_limiter
                             .until_n_ready(NonZeroU32::new(chunk.cardinality() as u32).unwrap())
                             .await;
-                        assert!(
-                            result.is_ok(),
-                            "the capacity of rate_limiter must be larger than the cardinality of chunk"
-                        );
+                        if let Err(InsufficientCapacity(n)) = result {
```

Comment:
May I ask the motivation of changing to error from panic? Although it looks good to me.

Reply:
It will crash the cluster, I think it is better to just emit a log instead.
Trade-off is of course if the user doesn't check the log, they could miss it.

A proper fix is to ensure when setting the `STREAMING_RATE_LIMIT` property, the rate limit is larger than chunk size.
But that requires a bit more refactoring work, since now the session config handler needs to be made aware of system configs as well. So we leave it for next time.